python/plugins/processing/algs/qgis/CreateAtlasGrid.py

```diff
@@ -0,0 +1,178 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+__author__ = 'Francisco Pérez Sampayo'
+__date__ = '2019-02-27'
+__copyright__ = '(C) 2019 by fpsampayo'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (QgsRectangle,
+                       QgsProject,
+                       QgsProcessingException,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingParameterBoolean,
+                       QgsProcessingParameterDistance,
+                       QgsProcessing,
+                       QgsFields,
+                       QgsField,
+                       QgsWkbTypes,
+                       QgsFeature,
+                       QgsGeometry)
+
+
+class CreateAtlasGrid(QgisAlgorithm):
+
+    LAYOUT = 'LAYOUT'
+    COVERAGE_LAYER = 'COVERAGE_LAYER'
+    COVERAGE_BUFFER = 'COVERAGE_BUFFER'
+    ONLY_ON_FEATURES = 'ONLY_ON_FEATURES'
+    OUTPUT = 'OUTPUT'
+
+    def name(self):
+        # Unique (non-user visible) name of algorithm
+        return 'createatlasgrid'
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Create a grid from map layout')
+
+    def group(self):
+        return self.tr('Cartography')
+
+    def groupId(self):
+        return 'cartography'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+
+        layout_param = QgsProcessingParameterString(
+            self.LAYOUT,
+            description=self.tr('Layout to use'),
+            defaultValue=None,
+            optional=False)
+
+        layout_param.setMetadata(
+            {'widget_wrapper': {
+                'class':
+                    'processing.gui.wrappers_layout.LayoutWrapper'}})
+        self.addParameter(layout_param)
+
+        self.addParameter(QgsProcessingParameterFeatureSource(self.COVERAGE_LAYER,
+                                                              self.tr('Coverage layer'),
+                                                              [QgsProcessing.TypeMapLayer]))
+        self.addParameter(QgsProcessingParameterDistance(self.COVERAGE_BUFFER,
```

Comment:
this admit negative value.. I suppose could make sense to have a internal coverage.

Reply:
I think that this has no sense(internal coverage) but maybe some user thinks different. This is the reason for allow negative inputs.